Ginger/GingerCoreNET/Database/NoSqlBase/GingerCassandra.cs

```diff
@@ -44,25 +46,62 @@ public override bool Connect()
+                string sslValue = null;
                 int queryTimeout = 20000;//default timeout (20 seconds).
-                if (Db.DatabaseOperations.TNSCalculated.ToLower().Contains(queryTimeoutString.ToLower()))
+                string[] queryArray = Db.DatabaseOperations.TNSCalculated.Split(';');
+                if (queryArray[1].ToLower().Contains(queryTimeoutString.ToLower()))
                 {
-                    string queryTimeoutValue = Db.DatabaseOperations.TNSCalculated.Substring(Db.DatabaseOperations.TNSCalculated.ToLower().IndexOf(queryTimeoutString.ToLower()) + queryTimeoutString.Length);
+                    string queryTimeoutValue = queryArray[1].Substring(queryArray[1].ToLower().IndexOf(queryTimeoutString.ToLower()) + queryTimeoutString.Length);
                     queryTimeout = Convert.ToInt32(queryTimeoutValue) * 1000;
                 }
 
-                string[] HostKeySpace = Db.DatabaseOperations.TNSCalculated.Split('/');
+                if (queryArray[2].ToLower().Contains(SLL.ToLower()))
+                {
+                    sslValue = queryArray[2].Substring(queryArray[2].ToLower().IndexOf(SLL.ToLower()) + SLL.Length);
+                }
+                string[] HostKeySpace = queryArray[0].Split('/');
                 string[] HostPort = HostKeySpace[0].Split(':');
-
+               
                 if (HostPort.Length == 2)
                 {
                     if (string.IsNullOrEmpty(Db.Pass) && string.IsNullOrEmpty(Db.User))
                     {
-                        cluster = Cluster.Builder().AddContactPoint(HostPort[0]).WithPort(Int32.Parse(HostPort[1])).WithQueryTimeout(queryTimeout).Build();
+                        if (string.IsNullOrEmpty(sslValue))
+                        {
+                            cluster = Cluster.Builder().AddContactPoint(HostPort[0]).WithPort(Int32.Parse(HostPort[1])).WithQueryTimeout(queryTimeout).Build();
+                        }
+                        else
+                        {
+                            var sslOptions = new SSLOptions(
+                                        Enum.Parse<SslProtocols>(sslValue), false,
+                                        (sender, certificate, chain, errors) => { return true; });
+                            cluster = Cluster.Builder()
+                                .AddContactPoint(HostPort[0])
+                                .WithPort(Int32.Parse(HostPort[1]))
+                                .WithSSL(sslOptions)
+                                .WithQueryTimeout(queryTimeout)
+                                .Build();
+                        }
                     }
                     else
                     {
-                        cluster = Cluster.Builder().WithCredentials(Db.User.ToString(), Db.Pass.ToString()).AddContactPoint(HostPort[0]).WithPort(Int32.Parse(HostPort[1])).WithQueryTimeout(queryTimeout).Build();
+                        if (string.IsNullOrEmpty(sslValue)) {
```

Comment:
comparing sslValue multiple times, so have it in a separate method to calculate SSL options.
Also nothing found to compare different values of SSL protocols to be used as we discussed we have none, ssl12, ssl1.0 etc. Also what if the SSL value passed does not fall under the allowed values in enum